qiskit/assembler/assemble_schedules.py

```diff
@@ -109,18 +107,34 @@ def assemble_schedules(schedules, qobj_id, qobj_header, run_config):
                         channel=instruction.channels[0])
                 # add samples to pulse library
                 user_pulselib[name] = instruction.command
-            elif isinstance(instruction, AcquireInstruction):
+
+            if isinstance(instruction, AcquireInstruction):
                 max_memory_slot = max(max_memory_slot,
                                       *[slot.index for slot in instruction.mem_slots])
+                acquire_instruction_map[(shift, instruction.command)].append(instruction)
+                continue
 
-                acquire_instructions.append(instruction)
-
-            converted_instruction = instruction_converter(shift, instruction)
-            qobj_instructions.append(converted_instruction)
+            if isinstance(instruction, DelayInstruction):
+                # delay instructions are ignored as timing is explicit within qobj
+                continue
 
-        if meas_map:
-            # verify all acquires satisfy meas_map
-            _validate_meas_map(acquire_instructions, meas_map)
+            qobj_instructions.append(instruction_converter(shift, instruction))
+
+        if acquire_instruction_map:
+            if meas_map:
+                _validate_meas_map(acquire_instruction_map, meas_map)
+            for (shift, _), instructions in acquire_instruction_map.items():
```

Comment:
Could this be broken into a function like `_validate_meas_map`?

Reply:
Sure

Reply:
also think I will refactor this file to make a few separate functions (e.g. assemble_instructions)

Reply:
not in this PR though